src/client.ts

```diff
@@ -199,6 +201,12 @@ export class Client {
 
     const tracerProvider = new NodeTracerProvider()
     tracerProvider.addSpanProcessor(new SpanProcessor(this))
+    if (this.config.data.testMode && this.config.data.testModeFilePath) {
+      const filePath = this.config.data.testModeFilePath
+      const spanProcessor = new TestModeSpanProcessor(filePath)
+
+      tracerProvider.addSpanProcessor(spanProcessor)
```

Comment:
I was a bit unsure what was tested here, because this doesn't use our real SpanProcessor. It's testing all the automatically added instrumentation up to the SpanProcessor part, right?

Like I mentioned in an earlier call, we can also call `toObject()` on an OpenTelemetry Span in the SpanProcessor when this test mode is active and write that extension parsed result to a file.

Reply:
I'll take some time to try the `toObject()` approach.

Reply:
No need to block the PR on it. You can look at it in another PR if you want.

Reply:
As talked earlier this week, I think we should keep this out of scope as the plans are relying on the upcoming HTTP server in the agent and get the spans from there.